src/libraries/System.Runtime.Numerics/src/System/Number.Polyfill.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    // Polyfill CoreLib internal interfaces and methods
+    // Define necessary members only
+
+    internal interface IUtfChar<TSelf>
```

Comment:
CoreLib comments indicate this interface to be a workaround for inliner limitation. Is it resolved now? Then we can move to `IBinaryInteger` and remove the polyfill.

Reply:
It looks like resolved with simple benchmarking on Int*.Parse:

| Method       | Job        | Toolchain                  | value                | Mean      | Error     | StdDev    | Median    | Min       | Max       | Ratio | RatioSD | Allocated | Alloc Ratio |
|------------- |----------- |--------------------------- |--------------------- |----------:|----------:|----------:|----------:|----------:|----------:|------:|--------:|----------:|------------:|
| Parse        | Job-MGZKSL | \BinaryInteger\corerun.exe | -9223372036854775808 | 10.932 ns | 0.1064 ns | 0.0996 ns | 10.935 ns | 10.747 ns | 11.130 ns |  1.02 |    0.02 |         - |          NA |
| Parse        | Job-TCKMSW | \base\corerun.exe          | -9223372036854775808 | 10.725 ns | 0.2329 ns | 0.2178 ns | 10.693 ns | 10.366 ns | 11.095 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| TryParse     | Job-MGZKSL | \BinaryInteger\corerun.exe | -9223372036854775808 | 11.103 ns | 0.1603 ns | 0.1499 ns | 11.035 ns | 10.923 ns | 11.374 ns |  1.02 |    0.03 |         - |          NA |
| TryParse     | Job-TCKMSW | \base\corerun.exe          | -9223372036854775808 | 10.898 ns | 0.2303 ns | 0.2464 ns | 10.810 ns | 10.653 ns | 11.524 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| ParseSpan    | Job-MGZKSL | \BinaryInteger\corerun.exe | -9223372036854775808 | 11.050 ns | 0.0883 ns | 0.0826 ns | 11.030 ns | 10.945 ns | 11.208 ns |  1.00 |    0.01 |         - |          NA |
| ParseSpan    | Job-TCKMSW | \base\corerun.exe          | -9223372036854775808 | 11.100 ns | 0.1081 ns | 0.1011 ns | 11.091 ns | 10.927 ns | 11.268 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| TryParseSpan | Job-MGZKSL | \BinaryInteger\corerun.exe | -9223372036854775808 | 10.851 ns | 0.1740 ns | 0.1453 ns | 10.859 ns | 10.575 ns | 11.110 ns |  1.01 |    0.01 |         - |          NA |
| TryParseSpan | Job-TCKMSW | \base\corerun.exe          | -9223372036854775808 | 10.698 ns | 0.0577 ns | 0.0512 ns | 10.692 ns | 10.621 ns | 10.786 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| Parse        | Job-MGZKSL | \BinaryInteger\corerun.exe | 12345                |  4.582 ns | 0.0285 ns | 0.0253 ns |  4.589 ns |  4.535 ns |  4.624 ns |  1.01 |    0.01 |         - |          NA |
| Parse        | Job-TCKMSW | \base\corerun.exe          | 12345                |  4.558 ns | 0.0157 ns | 0.0139 ns |  4.561 ns |  4.526 ns |  4.574 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| TryParse     | Job-MGZKSL | \BinaryInteger\corerun.exe | 12345                |  4.643 ns | 0.0555 ns | 0.0520 ns |  4.627 ns |  4.588 ns |  4.772 ns |  1.09 |    0.02 |         - |          NA |
| TryParse     | Job-TCKMSW | \base\corerun.exe          | 12345                |  4.265 ns | 0.0750 ns | 0.0701 ns |  4.232 ns |  4.184 ns |  4.407 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| ParseSpan    | Job-MGZKSL | \BinaryInteger\corerun.exe | 12345                |  4.233 ns | 0.0850 ns | 0.0795 ns |  4.206 ns |  4.137 ns |  4.371 ns |  0.95 |    0.02 |         - |          NA |
| ParseSpan    | Job-TCKMSW | \base\corerun.exe          | 12345                |  4.474 ns | 0.0274 ns | 0.0256 ns |  4.478 ns |  4.432 ns |  4.520 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| TryParseSpan | Job-MGZKSL | \BinaryInteger\corerun.exe | 12345                |  4.646 ns | 0.0341 ns | 0.0302 ns |  4.650 ns |  4.589 ns |  4.684 ns |  0.99 |    0.01 |         - |          NA |
| TryParseSpan | Job-TCKMSW | \base\corerun.exe          | 12345                |  4.679 ns | 0.0773 ns | 0.0723 ns |  4.646 ns |  4.611 ns |  4.800 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| Parse        | Job-MGZKSL | \BinaryInteger\corerun.exe | 9223372036854775807  | 10.146 ns | 0.0732 ns | 0.0685 ns | 10.185 ns |  9.996 ns | 10.207 ns |  1.01 |    0.01 |         - |          NA |
| Parse        | Job-TCKMSW | \base\corerun.exe          | 9223372036854775807  | 10.081 ns | 0.0829 ns | 0.0776 ns | 10.066 ns |  9.995 ns | 10.218 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| TryParse     | Job-MGZKSL | \BinaryInteger\corerun.exe | 9223372036854775807  |  9.959 ns | 0.0401 ns | 0.0335 ns |  9.971 ns |  9.865 ns |  9.997 ns |  0.99 |    0.01 |         - |          NA |
| TryParse     | Job-TCKMSW | \base\corerun.exe          | 9223372036854775807  | 10.099 ns | 0.0562 ns | 0.0499 ns | 10.096 ns | 10.044 ns | 10.204 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| ParseSpan    | Job-MGZKSL | \BinaryInteger\corerun.exe | 9223372036854775807  | 10.030 ns | 0.0773 ns | 0.0686 ns | 10.043 ns |  9.939 ns | 10.182 ns |  0.99 |    0.01 |         - |          NA |
| ParseSpan    | Job-TCKMSW | \base\corerun.exe          | 9223372036854775807  | 10.129 ns | 0.1174 ns | 0.0916 ns | 10.089 ns | 10.025 ns | 10.259 ns |  1.00 |    0.00 |         - |          NA |
|              |            |                            |                      |           |           |           |           |           |           |       |         |           |             |
| TryParseSpan | Job-MGZKSL | \BinaryInteger\corerun.exe | 9223372036854775807  | 10.054 ns | 0.0628 ns | 0.0588 ns | 10.059 ns |  9.957 ns | 10.166 ns |  0.96 |    0.02 |         - |          NA |
| TryParseSpan | Job-TCKMSW | \base\corerun.exe          | 9223372036854775807  | 10.503 ns | 0.2312 ns | 0.2270 ns | 10.460 ns | 10.150 ns | 11.025 ns |  1.00 |    0.00 |         - |          NA |

Reply:
Hmm `IBinaryInteger` may be unfriendly for mono.

Reply:
Yeah the issue was with Mono and its something on their backlog to get resolved.